components/browser/state/src/main/java/mozilla/components/browser/state/reducer/EngineStateReducer.kt

```diff
@@ -45,6 +45,9 @@ internal object EngineStateReducer {
         is EngineAction.OptimizedLoadUrlTriggeredAction -> {
             state
         }
+        is EngineAction.SaveToPdfErrorAction -> {
+            state
```

Comment:
I was looking at this kind of code recently as well (for thumbnails). 

I don't think we should return `state` here because this is almost like a silent failure that a consumer would not know _requires_ a middleware to handle this action. We can see below how the EngineActions there `throw` to notify the app that they need another piece to handle this case.

The question comes up then, why does `EngineAction.OptimizedLoadUrlTriggeredAction` right above this return `state` then? I believe it was because [it's a performance improvement](https://github.com/mozilla-mobile/android-components/issues/11066) and was explicitly intended to do nothing. (cc @csadilek in case I'm wrong here)

What do you think?

Reply:
Yes, exactly, OptimizedLoadUrlTriggeredAction doesn't do anything to the state. We have an optimized code path in SessionUseCases and it felt wrong to us at the time that this one path would not be observable on the store. So that's what this action is for, make it obervable that a page to was loaded in all cases. This is useful in Fenix here for instance: https://github.com/mozilla-mobile/fenix/blob/c7ebec360227c70ce2ba0243b88321eac39eee89/app/src/main/java/org/mozilla/fenix/historymetadata/HistoryMetadataMiddleware.kt#L98